pkg/hooks/types.go

```diff
@@ -0,0 +1,112 @@
+package hooks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/types"
+)
+
+// HookType represents the stage in the Hook lifecycle
+type HookType string
+
+func (hookType HookType) String() string {
+	return string(hookType)
+}
+
+var (
+	// BeforeInstall hooks run before the artifact is downloaded and extracted
+	BeforeInstall = HookType("before_install")
+	// AfterInstall hooks run after we have installed but before we have disabled the old version
+	AfterInstall = HookType("after_install")
+	// BeforeUninstall happens after shutdown but before uninstall
+	BeforeUninstall = HookType("before_uninstall")
+	// BeforeLaunch occurs after we have disabled the old version
+	BeforeLaunch = HookType("before_launch")
+	// AfterLaunch occurs after launch
+	AfterLaunch = HookType("after_launch")
+	// AfterAuth occurs conditionally when artifact authorization fails
+	AfterAuthFail = HookType("after_auth_fail")
+)
+
+func AsHookType(value string) (HookType, error) {
+	switch value {
+	case BeforeInstall.String():
+		return BeforeInstall, nil
+	case AfterInstall.String():
+		return AfterInstall, nil
+	case BeforeUninstall.String():
+		return BeforeUninstall, nil
+	case BeforeLaunch.String():
+		return BeforeLaunch, nil
+	case AfterLaunch.String():
+		return AfterLaunch, nil
+	case AfterAuthFail.String():
+		return AfterAuthFail, nil
+	default:
+		return HookType(""), fmt.Errorf("%s is not a valid hook type", value)
+	}
+}
+
+// DefaultPath is a directory on disk where hooks are installed by default.
+var DefaultPath = "/usr/local/p2hooks.d"
```

Comment:
should this be const? Or do we want to keep as var to allow override at build-time?